src/main/java/org/immregistries/mqe/validator/engine/rules/TypedValidationRule.java

```diff
@@ -0,0 +1,30 @@
+package org.immregistries.mqe.validator.engine.rules;
+
+import org.immregistries.mqe.validator.engine.ValidationRule;
+import org.immregistries.mqe.vxu.TargetType;
+
+public class TypedValidationRule<T extends ValidationRule> {
```

Comment:
I wonder if this is necessary. It implies there is the need for an untyped validation rule, which might be necessary, but I don't know why off the top of my head. Could we just add the type to the "validation rule" class and have the type and targetType be part of a constructor there?